File: ingest-rest/src/main/java/org/chronopolis/ingest/api/serializer/AceTokenSerializer.java
package org.chronopolis.ingest.api.serializer;

import com.fasterxml.jackson.core.JsonGenerator;
import com.fasterxml.jackson.databind.JsonSerializer;
import com.fasterxml.jackson.databind.SerializerProvider;
import org.chronopolis.rest.entities.AceToken;
import org.chronopolis.rest.models.AceTokenModel;

import java.io.IOException;
import java.time.ZoneOffset;
import java.time.ZonedDateTime;

/**
 * Serializer to return an {@link AceTokenModel} from an {@link AceToken} entity
 * Use a UTC zone for the create ZonedDateTime
 *
 * @author shake
 */
public class AceTokenSerializer extends JsonSerializer<AceToken> {
    @Override
    public void serialize(AceToken value, JsonGenerator gen, SerializerProvider serializers) throws IOException {
        AceTokenModel model = new AceTokenModel();
        model.setId(value.getId())
             .setBagId(value.getBag().getId())
             .setRound(value.getRound())
             .setProof(value.getProof())
             .setAlgorithm(value.getAlgorithm())
             .setFilename(value.getFilename())
             .setImsService(value.getImsService())
             .setCreateDate(ZonedDateTime.ofInstant(value.getCreateDate().toInstant(), ZoneOffset.UTC));
        gen.writeObject(model);
    }
}

File: rest-entities/src/main/kotlin/org/chronopolis/rest/entities/serializers/AceTokenSerializer.kt
package org.chronopolis.rest.entities.serializers

import com.fasterxml.jackson.core.JsonGenerator
import com.fasterxml.jackson.databind.JsonSerializer
import com.fasterxml.jackson.databind.SerializerProvider
import org.chronopolis.rest.models.AceToken
import org.chronopolis.rest.entities.AceToken as AceTokenEntity

/**
 * Serializer to return an [AceToken] model from an [AceTokenEntity] entity
 *
 * Uses the UTC-0 zone for the create date's ZonedDateTime
 *
 * @author shake
 */
class AceTokenSerializer : JsonSerializer<AceTokenEntity>() {
    override fun serialize(token: AceTokenEntity,
                           gen: JsonGenerator,
                           serializers: SerializerProvider) {
        gen.writeObject(token.model())
    }
}